arlo-client/src/components/AuditFlow/Ballot.test.tsx

```diff
@@ -0,0 +1,195 @@
+          ballots={dummyBallots.ballots}
+          boardName="audit board #1"
+          contest="contest name"
+          previousBallot={jest.fn()}
+          nextBallot={jest.fn()}
+          submitBallot={jest.fn()}
+          roundIx="1"
+          batchId="batch-id-1"
+          ballotId={313}
+        />
+      </Router>
+    )
+    expect(container).toMatchSnapshot()
+  })
+
+  it('switches audit and review views', async () => {
+    const { container, getByText, getByTestId } = render(
+      <Router history={history}>
+        <Ballot
+          home="/election/1/board/1"
+          ballots={dummyBallots.ballots}
+          boardName="audit board #1"
+          contest="contest name"
+          previousBallot={jest.fn()}
+          nextBallot={jest.fn()}
+          submitBallot={jest.fn()}
+          roundIx="1"
+          batchId="batch-id-1"
+          ballotId={313}
+        />
+      </Router>
+    )
+
+    fireEvent.click(getByTestId('YES'), { bubbles: true })
+    await wait(() =>
+      fireEvent.click(getByTestId('enabled-review'), { bubbles: true })
+    )
+    await wait(() => {
+      expect(getByText('Submit & Next Ballot')).toBeTruthy()
+    })
+    await wait(() => {
+      expect(container).toMatchSnapshot()
+    })
+    fireEvent.click(getByText('Edit'), { bubbles: true })
+    await wait(() => {
+      expect(getByText('Review')).toBeTruthy()
+    })
+  })
+
+  it('toggles and submits comment', async () => {
+    const { container, getByText, getByTestId } = render(
+      <Router history={history}>
+        <Ballot
+          home="/election/1/board/1"
+          ballots={dummyBallots.ballots}
+          boardName="audit board #1"
+          contest="contest name"
+          previousBallot={jest.fn()}
+          nextBallot={jest.fn()}
+          submitBallot={jest.fn()}
+          roundIx="1"
+          batchId="batch-id-1"
+          ballotId={313}
+        />
+      </Router>
+    )
+
+    const commentInput = getByTestId('comment-textarea')
+    fireEvent.change(commentInput, { target: { value: 'a test comment' } })
+
+    fireEvent.click(getByTestId('YES'), { bubbles: true })
+    await wait(() =>
+      fireEvent.click(getByTestId('enabled-review'), { bubbles: true })
+    )
+    await wait(() => {
+      expect(getByText('Submit & Next Ballot')).toBeTruthy()
+      expect(getByText('COMMENT: a test comment')).toBeTruthy()
+      expect(container).toMatchSnapshot()
+    })
+  })
+
+  it('submits review and progresses to next ballot', async () => {
+    const submitMock = jest.fn()
+    const nextBallotMock = jest.fn()
+    const { container, getByText, getByTestId } = render(
+      <Router history={history}>
+        <Ballot
+          home="/election/1/board/1"
+          ballots={dummyBallots.ballots}
+          boardName="audit board #1"
+          contest="contest name"
+          previousBallot={jest.fn()}
+          nextBallot={nextBallotMock}
+          submitBallot={submitMock}
+          roundIx="1"
```

Comment:
This is short for `roundIndex`, not `roundId`. The id for the round is not useful for displaying to users since it's not readable, but the index _is_ displayed.